src/sidebar/components/new-note-btn.js

```diff
@@ -1,25 +1,31 @@
 import { createElement } from 'preact';
 import propTypes from 'prop-types';
 
-import events from '../events';
+import uiConstants from '../ui-constants';
 import useStore from '../store/use-store';
 import { withServices } from '../util/service-context';
 import { applyTheme } from '../util/theme';
 
 import Button from './button';
 
-function NewNoteButton({ $rootScope, settings }) {
-  const store = useStore(store => ({
-    frames: store.frames(),
-  }));
+function NewNoteButton({ settings }) {
+  const frames = useStore(store => store.frames());
+  const isLoggedIn = useStore(store => store.isLoggedIn());
+
+  const createAnnotation = useStore(store => store.createAnnotation);
+  const openSidebarPanel = useStore(store => store.openSidebarPanel);
 
   const onNewNoteBtnClick = function() {
-    const topLevelFrame = store.frames.find(f => !f.id);
+    if (!isLoggedIn) {
+      openSidebarPanel(uiConstants.PANEL_LOGIN_PROMPT);
+      return;
+    }
+    const topLevelFrame = frames.find(f => !f.id);
     const annot = {
       target: [],
       uri: topLevelFrame.uri,
     };
-    $rootScope.$broadcast(events.BEFORE_ANNOTATION_CREATED, annot);
+    createAnnotation(annot);
```

Comment:
I found one place in the app that references the `BEFORE_ANNOTATION_CREATED` event which doesn't just call `store.createAnnotation()`. That event handler does two things which we should check either still happen or are not relevant:

1. Scroll the new note or annotation into view in the sidebar
2. Clear the selection, if the user has made a selection by clicking on a highlight in the page.

Reply:
I tested this manually just now:

1. Clicking the "New Note" button on this branch does result in the view being scrolled to the new note. I suspect this may be happening just as a side effect of the annotation body input field being focused.
2. The "Clearing the selection" part is irrelevant because in order to see this button, you have to switch from the Annotations tab to the Notes tab which will clear the selection anyway

Reply:
I was just looking at this, too! Your initial point is still relevant, however, as we'll have to refactor this code in the near term, so I'm walking through the various implications.